src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/AwsBedrockEnbeddingProvider.java

```diff
@@ -0,0 +1,102 @@
+package io.stargate.sgv2.jsonapi.service.embedding.operation;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.EmbeddingProviderConfigStore;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.ProviderConstants;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.bedrockruntime.BedrockRuntimeAsyncClient;
+import software.amazon.awssdk.services.bedrockruntime.model.InvokeModelResponse;
+
+/** Provider implementation for AWS Bedrock. To start we support only Titan embedding models. */
+public class AwsBedrockEnbeddingProvider extends EmbeddingProvider {
+
+  private static final String providerId = ProviderConstants.BEDROCK;
+  private static ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+  private static ObjectReader or = new ObjectMapper().reader();
+
+  public AwsBedrockEnbeddingProvider(
+      EmbeddingProviderConfigStore.RequestProperties requestProperties,
+      String baseUrl,
+      String modelName,
+      int dimension,
+      Map<String, Object> vectorizeServiceParameters) {
+    super(
+        requestProperties,
+        baseUrl,
+        modelName,
+        acceptsTitanAIDimensions(modelName) ? dimension : 0,
+        vectorizeServiceParameters);
+  }
+
+  @Override
+  public Uni<Response> vectorize(
+      int batchId,
+      List<String> texts,
+      EmbeddingProvider.Credentials credentials,
+      EmbeddingRequestType embeddingRequestType) {
+    if (!credentials.accessKeyId().isPresent() || !credentials.secretAccessKey().isPresent()) {
+      throw new RuntimeException("AccessId and SecretId are required for AWS Bedrock");
```

Comment:
Not sure how we do it with other clients; is it not possible to throw `JsonApiException`?

Reply:
Yes., will change it. Wanted the first version to work, resolving this.